backends-common/postgres/src/main/java/org/apache/james/backends/postgres/quota/PostgresQuotaCurrentValueDAO.java

```diff
@@ -0,0 +1,120 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.postgres.quota;
+
+import static org.apache.james.backends.postgres.quota.PostgresQuotaModule.PostgresQuotaCurrentValueTable.COMPONENT;
+import static org.apache.james.backends.postgres.quota.PostgresQuotaModule.PostgresQuotaCurrentValueTable.CURRENT_VALUE;
+import static org.apache.james.backends.postgres.quota.PostgresQuotaModule.PostgresQuotaCurrentValueTable.IDENTIFIER;
+import static org.apache.james.backends.postgres.quota.PostgresQuotaModule.PostgresQuotaCurrentValueTable.PRIMARY_KEY_CONSTRAINT_NAME;
+import static org.apache.james.backends.postgres.quota.PostgresQuotaModule.PostgresQuotaCurrentValueTable.TABLE_NAME;
+import static org.apache.james.backends.postgres.quota.PostgresQuotaModule.PostgresQuotaCurrentValueTable.TYPE;
+
+import java.util.function.Function;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaCurrentValue;
+import org.apache.james.core.quota.QuotaType;
+import org.jooq.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PostgresQuotaCurrentValueDAO {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresQuotaCurrentValueDAO.class);
+
+    private final PostgresExecutor postgresExecutor;
+
+    public PostgresQuotaCurrentValueDAO(PostgresExecutor postgresExecutor) {
+        this.postgresExecutor = postgresExecutor;
+    }
+
+    public Mono<Void> increase(QuotaCurrentValue.Key quotaKey, long amount) {
+        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.insertInto(TABLE_NAME)
+            .set(IDENTIFIER, quotaKey.getIdentifier())
+            .set(COMPONENT, quotaKey.getQuotaComponent().getValue())
+            .set(TYPE, quotaKey.getQuotaType().getValue())
+            .set(CURRENT_VALUE, amount)
+            .onConflictOnConstraint(PRIMARY_KEY_CONSTRAINT_NAME)
```

Comment:
Oh... does that allow you to try an insert and fallback on an update if already exists? Could not find that (honestly Jooq doc is terrible) and I'm happy to see this, will be useful with annotations and the hstore :)

Reply:
Yes. Thanks to ChatGPT helped me and @hungphan227 on this yesterday :))

ChatGPT can help to write the SQL queries and translate them to jooq code :D

![image](https://github.com/apache/james-project/assets/55171818/024175ad-f3e0-41a9-a818-a80089d55343)

Reply:
Damn does everybody but me has an account now? :rofl: ok thanks :)

Reply:
I do not have it too. I am considering getting one. It is helpful :))

Reply:
Please also ask him to be sure this is an atomic PG query -> No transaction involved?

Reply:
![image](https://github.com/apache/james-project/assets/55171818/342d7e33-0ed0-4d5f-9d21-eac1d362a376)

It is expected though: every SQL query is an auto-commit transaction.